document/workflow-detail-level.service.ts

```diff
@@ -42,6 +48,17 @@ export class WorkflowDetailLevelService {
     private readonly _showFQN: BehaviorSubject<boolean>;
 
 
+    /**
+     * holds the mapping mode state for each mapping by its ID
+     */
+    private readonly _mappingModes = new Map<string, MappingMode>();
+
+    /**
+     * Subject to trigger if mappingMode state changes for an ID
+     */
+    private readonly _mappingModeChange = new Subject<string>();
```

Comment:
why not combine these two and use a BahaviorSubject?

Reply:
This way it works like the collapse state.